_filters/execute_magick.py

```diff
@@ -0,0 +1,133 @@
+#!/bin/env python
+
+"""
+Pandoc filter to run all imagemagick code in code blocks
+"""
+
+import os
+import sys
+import hashlib
+from pandocfilters import toJSONFilter, Image, CodeBlock, Para
+from subprocess import Popen, PIPE
+
+DEBUG = False
+
+IMAGEDIR = '_images'
+
+# List of imagemagick tools; only commands in this list will be allowed to be run by the filter
+IM_COMMANDS = ('animate', 'compare', 'composite', 'conjure', 'convert', 'display', 'identify', 'import', 'mogrify', 'montage', 'stream', 'echo', 'printf')
+IM_IMAGE_TYPES = ('png', 'jpg', 'gif', 'tif')
+
+# Needed for writing unicode to stdout/stderr:
+reload(sys)
+sys.setdefaultencoding('utf-8')
+
+def sha1(x):
+    return hashlib.sha1(x.encode(sys.getfilesystemencoding())).hexdigest()
+
+def execute(code):
+    """
+    Executes the code.
+    Performs some sanity checks first.
+    """
+#    if DEBUG:
+#        sys.stderr.write('-'*100 + '\n')
+    code = remove_slash(code)
+    for line in code:
+        if DEBUG:
+            sys.stderr.write(line + u'\n')
+        commandline = line.split()
+        if commandline[0] not in IM_COMMANDS:
```

Comment:
[YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it). We can assume all flagged are intended for shell consumption. No need to white-list anything.